doc/git/providers.mkd

```diff
@@ -50,6 +50,24 @@ git:
   username: 'git'
 ```
 
+#### Supported transports with Rugged
+
+For ssh support, you need to have libssh2 installed (along with the relevant dev package/headers) before you install the Rugged gem.
+
+For https support on Linux, you need to have OpenSSL installed (along with the relevant dev package/headers) before you install the Rugged gem. OS X and Windows support should automatically include https support.
+
+You can check which features your Rugged supports by using the following in irb:
+
+```ruby
+irb(main):001:0> require('rugged')
+=> true
+irb(main):002:0> Rugged.features
+=> [:threads, :https, :ssh]
```

Comment:
r10k 2.0.0 will automatically check for the :https and :ssh features and will warn if libgit2/rugged was compiled without them: https://github.com/puppetlabs/r10k/commit/94bfeebc1e8f4a723187c60c78ab5fcde2e1bd39